x-pack/plugin/sql/qa/server/src/main/resources/math.csv-spec

```diff
@@ -6,175 +6,192 @@ SELECT TRUNC(ASCII(LEFT(first_name, 1)), 1), COUNT(*) count FROM test_emp GROUP
 
 TRUNC(ASCII(LEFT(first_name, 1)), 1):i   |     count:l
 -----------------------------------------+---------------
-null                                     |10             
```

Comment:
🔥

Reply:
> I'm also wondering wether there might be an implementation for integer types that can avoid the overflow. Something like `n - (n % (long)Math.pow(10, precision))`.

Unfortunately the overflow is not in the implementation, but in the semantics of the operation. Eg. 
`ROUND(9223372036854775807, -3)` is `9223372036854776000`, that is bigger than `Long.MAX_VALUE`. 

So we cannot do much, we just have to return Double (or fail, but it doesn't seem very convenient).

Reply:
Ah, got it 👍 

I would suggest to fail with an overflow error then instead of coercing into a double. That would also be consistent with the other math operators (I hope). E.g. `select 9223372036854775807 + 1` fails with `ApiError(500, 'arithmetic_exception', 'long overflow')`

Reply:
Actually, this approach could be considered for Float/Double values. 
I just have to figure out how precise it is with the half up/down rounding compared to the Math.round() and BigDecimal solutions.